api/src/middleware/team.js

```diff
@@ -12,15 +13,61 @@ async function validateTeamId(req, res, next) {
     .select('_id teamName orgName email teamMembers');
 
   if (foundTeam == null) {
-    next(
+    return next(
       fillErrorObject(404, 'Validation error', [
         'No team found with the given id',
       ]),
     );
   }
 
   req.foundTeam = foundTeam; // todo: does this need to be set inside middleware?
-  next();
+  return next();
+}
+
+async function validateGhUser(req, res, next) {
+  const { ghToken } = req.body;
+  const { data } = await axios.get('https://api.github.com/user',
+    {
+      headers: { Authorization: `token ${ghToken}` },
+    });
+  if (data.errors) {
+    return next(
+      fillErrorObject(400, 'Validation error: user doesnt exist!', [data.errors[0].detail]),
+    );
+  }
+  req.username = data.login;
+  return next();
+}
+
+async function validateTeamRepo(req, res, next) {
+  // Creating repoName
+  const { ghToken } = req.body;
+
+  const ghUsername = req.username;
+
+  const repoName = `${ghUsername}.github.io`;
+  try {
+    const repoValidator = await axios.get(`https://api.github.com/repos/${ghUsername}/${repoName}`, {
+      headers: {
+        Authorization: `token ${ghToken}`,
+        Accept: 'application/vnd.github.v3+json',
+      },
+    });
+    if (repoValidator.status !== 200) {
+      return next(
+        fillErrorObject(404, 'GH pages not found!', [
+          'GitHub Repo doesnt exist for this team!',
+        ]),
+      );
+    }
+    return next();
+  } catch (error) {
+    return next(
+      fillErrorObject(500, 'Server Error!', [
+        'Failed to access the repository!',
+      ]),
+    );
+  }
```

Comment:
When I clear data without the repo being made, it goes into the catch block so it's not going through L56-62. Might want to double check this logic.